### DISCUSSION TITLE
Add Mod Panel to User Profiles + a Handful of Changes

### DISCUSSION THREAD

**Discussion by michael-tharrington** [forem/forem#17311] [Icebox]
I'd love it if we could add the mod panel (currently only accessible on article pages) to user profiles and then make a few changes...

First off, here is what the Mod Panel looks like on an article:

![The layout of the Mod Panel + the shield icon](https://user-images.githubusercontent.com/16007075/163678255-a2906b3d-66a6-45b9-88c5-22e028ad1fb0.png)

## Requested Changes

There are a variety of changes we'd need to make in order to make this happen...

### 1. As part of moving the mod panel to user profiles, we'd need the shield icon to access the panel.**

The shield icon would need to appear somewhere on the user profile page and its functionality would mirror how it works on the article page, clicking it would make the mod panel appear

![Where the shield icon might appear...](https://user-images.githubusercontent.com/16007075/163677651-e7e89526-968a-4cbf-b7b2-dde1d22b4369.png)

### 2. We'd want to move the flag user action from `...` to being under the mod panel

Currently, "Flag User" can be found in the `...` menu and is collocated with other non-mod/non-admin actions.

I think we want this to be located in our mod panel and appear in a prominentspot near the top and available to all mods. It could potentially have a 🚩 emoji similar to how on the article mod panel we have emojis for 👍 high quality/ 👎 low quality and 🧐 flag to admins. I think we'd want the design here to be similar to the article mod page.

![Showing how the flag user button currently appears under the ... menu](https://user-images.githubusercontent.com/16007075/163678121-f899fd67-e433-4853-8b97-d5c3390a64a5.png)

### 3. We'd need an "OTHER THINGS YOU CAN DO" section that includes Admin actions 

It'd be fantastic if we could add an Admin actions section and include `Suspend`, `Comment Suspend` `Unpublish all Posts`, and `Banish`. Not only that, but perhaps buttons to make the community member a trusted user or even give them tag mod privileges. I think this would greatly improve the experience for an admin, as they could stay on site and access common actions while in their flow.

We could also potentially add functionality for moderators to `Suggest a Mod` and perhaps more...

![Admin actions section as it appears on articles](https://user-images.githubusercontent.com/16007075/163678949-e61899e8-1a0e-41af-86e3-58162e54943b.png)

## Why I want this

When reviewing spam and abusive behavior, I am frequently flipping between the community member-facing side of the site and the admin section and that constant switch can be a bit disruptive. It sounds silly but the page layout changes each time and I have to adjust my eyes, shift my mouse, etc. 

I really feel like I would get a lot out of having the ability to take actions from the panel in the moment of reviewing what I'm looking at on the screen. I've seriously loved having the ability to unpublish and feature articles from the mod panel and I just want to extend the use here to other things that I frequently do so that I can stay on the community-facing side of the community that I'm working in. 

I feel like there's more potential to this idea, I'm just not entirely sure what yet. I also think it pairs well with all the awesome updates we've been making to the admin UI. I honestly feel like these updates can coexist and we can have multiple ways for admin to take action both in and outside of the admin area depending on what flow we are in. 

## What do y'all think?

I'm eager to hear from others! What do y'all think about this idea?

**Comment by amywtlin** [comment]
@anujbhavsar96 Check this out - highly relevant to the `moderator` role we are working on.

**Reply by anujbhavsar96** [reply]
This is really great, @michael-tharrington . I love this idea of let admins perform actions from the user-facing side. I've talked to Jennie about this a few weeks back, like why always send admins to the admin portal to perform any action where they can perform from the user-facing side. And this request is right on point.

So this way, we will have a contextual mod panel (For articles and user profiles).

**Reply by michael-tharrington** [reply]
Aww yeah! So glad to hear that you dig this idea @anujbhavsar96. Let me know if there's any way I might be able to help with this.

**Reply by amywtlin** [reply]
@michael-tharrington Need to note that I read it a bit too fast and thought it was about our current mod panel on post view. This request to add it to user profile still makes sense to me, but I am going to have to icebox this one for now.

**Reply by michael-tharrington** [reply]
@amywtlin No worries at all! Totally cool to wait on this. I know y'all got loads going on and I do see this one as a nice-to-have more than a need-to-have-now.
